deeplearning4j-nn/src/main/java/org/deeplearning4j/datasets/iterator/AsyncDataSetIterator.java

```diff
@@ -60,14 +40,14 @@ public AsyncDataSetIterator(DataSetIterator iterator, int queueSize, BlockingQue
         this(iterator, queueSize, queue, true);
     }
 
-    /**
-     * Create an AsyncDataSetIterator with a queue size of 1 (i.e., only load a
-     * single additional DataSet)
-     *
-     * @param iterator
-     * @param queueSize
-     * @param queue BlockingQueue instance that will be used as backing queue. MagicQueue probably?
-     */
+    public AsyncDataSetIterator(DataSetIterator baseIterator, int queueSize) {
+        this(baseIterator, queueSize, new LinkedBlockingQueue<DataSet>(queueSize));
+    }
+
```

Comment:
Sure, this reasons is learnable size. We want workspace to learn memory requirements of a given DataSetIterator, and match that requirements. It's simple learning at this moment, but i'll improve it today, as discussed with @AlexDBlack earlier.